load_balancer/src/main.rs

```diff
@@ -0,0 +1,36 @@
+use std::net::SocketAddr;
+
+use hyper::server::conn::http1;
+
+use hyper_util::{rt::TokioIo, service::TowerToHyperService};
+use tokio::net::TcpListener;
+use tower::ServiceBuilder;
+
+use crate::{middleware::Logger, services::LoadBalancer};
+
+pub mod middleware;
+pub mod services;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
You can improve your development experience from the very beginning by using an error handling library such as color-eyre.